app/scripts/components/sandbox/mdxEditor/MDX.tsx

```diff
@@ -0,0 +1,264 @@
+import { generalErrorMessage } from '../../common/blocks/block-constant';
+import { MDX_LOCAL_STORAGE_KEY, MDX_SOURCE_DEFAULT } from '.';
+import { BlockComponent } from '$components/common/blocks';
+import { HintedErrorDisplay, docsMessage } from '$utils/hinted-error';
+
+const DEFAULT_CONTENT = '(your content here)';
+
+const DraggableEditor = styled.div`
+  position: absolute;
+  width: 800px;
+  z-index: 999;
+  right: 10px;
+`;
+
+const TitleBar = styled.div`
+  background-color: #1e1e1e;
+  color: #fff;
+  padding: 10px;
+  font-size: 14px;
+  font-weight: bold;
+  border-bottom: 1px solid #000;
+  cursor: move;
+`;
+
+const ResetButton = styled.button`
+  opacity: 0.5;
+  background-color: transparent;
+  text-decoration: underline;
+  color: #fff;
+  border: none;
+  font-size: 10px;
+  font-weight: normal;
+  cursor: pointer;
+`;
+
+const ErrorBar = styled.div`
+  background-color: #fff;
+  border: 3px solid ${themeVal('color.danger')};
+  color: ${themeVal('color.danger')};
+  padding: 4px;
+`;
+
+const EditorWrapper = styled.div`
+  overflow: scroll;
+  max-height: 500px;
+`;
+
+const GlobalErrorWrapper = styled.div`
+  border: 3px solid ${themeVal('color.danger')};
+  color: ${themeVal('color.danger')};
+  font-size: 14px;
+`;
+
+interface useMDXReturnProps {
+  source: string;
+  result: MDXContent | null;
+  error: any;
+}
+
+function useMDX(source) {
+  const [state, setState] = useState<useMDXReturnProps>({
+    source,
+    result: null,
+    error: null
+  });
+
+  async function setSource(source) {
+    const remarkPlugins = [remarkGfm];
+
+    let result: MDXContent | null = null;
+
+    try {
+      result = (
+        await evaluate(source, {
+          ...runtime,
+          useMDXComponents,
+          useDynamicImport: true,
+          remarkPlugins
+        } as any)
+      ).default;
+
+      setState((oldState) => {
+        return { ...oldState, source, result, error: null };
+      });
+    } catch (error) {
+      // console.log(error)
+      setState((oldState) => {
+        return { ...oldState, source, result: null, error };
+      });
+    }
+  }
+
+  useMemo(() => setSource(source), [source]);
+
+  return { ...state, setSource };
+}
+
+const MDXContext = createContext<MDXContent | null>(null);
+
+interface MDXEditorProps {
+  initialSource: string;
+  components: any;
+}
+
+const MDXEditor = ({ initialSource, components = null }: MDXEditorProps) => {
+  const { result, error, setSource } = useMDX(initialSource);
+  const [, setMdxSource] = useLocalStorage(
+    MDX_LOCAL_STORAGE_KEY,
+    MDX_SOURCE_DEFAULT
+  );
+
+  const extensions = useMemo<Extension[]>(
+    () => [basicSetup, oneDark, langMarkdown()],
+    []
+  );
+
+  const errorHumanReadable = useMemo(() => {
+    if (!error) return null;
+    const { line, message } = JSON.parse(JSON.stringify(error));
+    return `At line ${line - 1}: ${message}`;
+  }, [error]);
+
+  return (
+    <ErrorBoundaryWithCRAReset FallbackComponent={GlobalError}>
```

Comment:
What kind of errors trigger `GlobalError` component to show up? I tried to trigger it, but I couldn't find the way.

Reply:
None _that I am aware of_ ... But considering there's an `eval` happening within this ErrorBoundary, anything could happen (as in [zombo.com](https://zombo.com/)).
You can however manually trigger the error by commenting [this line](https://github.com/NASA-IMPACT/veda-ui/pull/551/files#diff-36308502277c3029ba212de372c8d8cb28d5f03c35c7c20d936c1d45aab9226fR174) out, and clear the editor's content.